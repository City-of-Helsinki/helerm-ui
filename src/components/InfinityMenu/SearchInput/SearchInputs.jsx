/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Select from 'react-select';

import SearchInput from './SearchInput';

export const FILTER_CONDITION_OPTIONS = [
  { value: 'and', label: 'JA' },
  { value: 'or', label: 'TAI' },
];

const SearchInputs = ({
  headerProps,
  isDetailSearch,
  searchInputs,
  filterCondition,
  addSearchInput,
  setSearchInput,
  removeSearchInput,
  onFilterConditionChange,
}) =>
  searchInputs.map((input, index) => (
    <div
      // eslint-disable-next-line react/no-array-index-key
      key={index}
      className={classNames({
        'col-xs-12 filters filters-detail-search-input': isDetailSearch,
        'col-sm-6': !isDetailSearch,
      })}
    >
      <SearchInput
        {...headerProps}
        searchInput={input}
        placeholder='Etsi...'
        setSearchInput={(event) => setSearchInput(index, event.target.value)}
      />
      {isDetailSearch && (
        <div className='filters-detail-search-input-buttons'>
          {index + 1 < searchInputs.length && (
            <Select
              className='Select'
              autoBlur
              isDisabled={index > 0}
              placeholder='Ehto'
              value={FILTER_CONDITION_OPTIONS.find(({ value }) => value === filterCondition)}
              isClearable={false}
              options={FILTER_CONDITION_OPTIONS}
              onChange={onFilterConditionChange}
            />
          )}
          <button
            type='button'
            className='btn btn-info btn-sm'
            onClick={() => removeSearchInput(index)}
            title='Poista hakuehto'
            aria-label='Poista hakuehto'
          >
            <span className='fa-solid fa-minus' aria-hidden='true' />
          </button>
          {index + 1 === searchInputs.length && (
            <button
              type='button'
              className='btn btn-info btn-sm'
              onClick={addSearchInput}
              title='Lis채채 hakuehto'
              aria-label='Lis채채 hakuehto'
            >
              <span className='fa-solid fa-plus' aria-hidden='true' />
            </button>
          )}
        </div>
      )}
    </div>
  ));

SearchInputs.propTypes = {
  headerProps: PropTypes.object,
  isDetailSearch: PropTypes.bool,
  searchInputs: PropTypes.array.isRequired,
  filterCondition: PropTypes.string.isRequired,
  addSearchInput: PropTypes.func.isRequired,
  setSearchInput: PropTypes.func.isRequired,
  removeSearchInput: PropTypes.func.isRequired,
  onFilterConditionChange: PropTypes.func.isRequired,
};

export default SearchInputs;
