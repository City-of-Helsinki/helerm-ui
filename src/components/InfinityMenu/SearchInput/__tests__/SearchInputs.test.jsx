import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import React from 'react';

import storeCreator from '../../../../store/createStore';
import SearchInputs, { FILTER_CONDITION_OPTIONS } from '../SearchInputs';

const baseMocks = {
  onFilterConditionChange: jest.fn(),
};

const renderComponent = (mocks = baseMocks) => {
  const history = createBrowserHistory();
  const store = storeCreator(history, {});

  return render(
    <Provider store={store}>
      <Router history={history}>
        <SearchInputs
          headerProps={{}}
          isDetailSearch
          searchInputs={['']}
          filterCondition={FILTER_CONDITION_OPTIONS[0].value}
          addSearchInput={jest.fn()}
          setSearchInput={jest.fn()}
          removeSearchInput={jest.fn()}
          onFilterConditionChange={mocks.onFilterConditionChange}
        />
      </Router>
    </Provider>,
  );
};

describe('<SearchInputs />', () => {
  it('renders correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
