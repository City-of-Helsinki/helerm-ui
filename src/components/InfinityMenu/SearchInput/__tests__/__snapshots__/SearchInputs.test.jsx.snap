// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInputs /> renders correctly 1`] = `
<div>
  <div
    class="col-xs-12 filters filters-detail-search-input"
  >
    <input
      class="react-infinity-menu-default-search-input"
      placeholder="Etsi..."
      type="search"
      value=""
    />
    <div
      class="filters-detail-search-input-buttons"
    >
      <button
        class="btn btn-info btn-sm"
        title="Poista hakuehto"
        type="button"
      >
        <span
          aria-hidden="true"
          class="fa-solid fa-minus"
        />
      </button>
      <button
        class="btn btn-info btn-sm"
        title="Lisää hakuehto"
        type="button"
      >
        <span
          aria-hidden="true"
          class="fa-solid fa-plus"
        />
      </button>
    </div>
  </div>
</div>
`;
